level_generator/level_generator.py

```diff
@@ -0,0 +1,178 @@
+
+    
+    #Sníží probabilitu o 1 prob. decrease
+    probability[last_direction.value] -= Config_lvl_gen.PROB_DECREASE
+
+    #Přidá probability do 2 zbývajících směrů (směry kolmé na předchozí)
+    x, y = last_direction.value
+    if (y, x) in pref_directions:
+        probability[(y, x)] += (Config_lvl_gen.PROB_DECREASE // Config_lvl_gen.PROB_DIVISOR)*(Config_lvl_gen.PROB_NUMERATOR)
+        probability[(-y, -x)] += (Config_lvl_gen.PROB_DECREASE // Config_lvl_gen.PROB_DIVISOR)*(Config_lvl_gen.PROB_DIVISOR - Config_lvl_gen.PROB_NUMERATOR)
+    else:
+        probability[(y, x)] += (Config_lvl_gen.PROB_DECREASE // Config_lvl_gen.PROB_DIVISOR)*(Config_lvl_gen.PROB_DIVISOR-Config_lvl_gen.PROB_NUMERATOR)
+        probability[(-y, -x)] += (Config_lvl_gen.PROB_DECREASE // Config_lvl_gen.PROB_DIVISOR)*(Config_lvl_gen.PROB_NUMERATOR)
+
+    #Preferovat smery od startu, rozdelit na mensi casti
+
+def pref_directions(start_point:tuple[int]) -> list[Direction]:
+    """"Return preferable directions from start point"""
+    x, y = start_point
+    pref_directions =[]
+    if x > Tile.WIDTH//2:
+        pref_directions.append(Direction.LEFT.value)
+    else:
+        pref_directions += [Direction.RIGHT.value]
+    if y > Tile.HEIGHT//2:
+        pref_directions += [Direction.UP.value]
+    else:
+        pref_directions += [Direction.DOWN.value]
+
+    return pref_directions
+
+def draw_point(x:int, y:int, img:Image)->None:
+    """Create 1 point on a specific pixel on the image"""
+    img.putpixel((x, y), Colors.PATH)
+
+def my_random_direction(probability:dict) -> Direction:
+    """Return a random direction based on probabilities"""
+    vectors_probability = []
+    for i in range(probability[Direction.UP.value]):
+        vectors_probability.append(Direction.UP)
+    for i in range(probability[Direction.DOWN.value]):
+        vectors_probability.append(Direction.DOWN)
+    for i in range(probability[Direction.RIGHT.value]):
+        vectors_probability.append(Direction.RIGHT)
+    for i in range(probability[Direction.LEFT.value]):
+        vectors_probability.append(Direction.LEFT)
+    
+    return random.choice(vectors_probability)
+
+def move_to(rnd_direction:Direction, x:int, y:int) -> tuple:
+    """Change x, y to a new value based on the Direction"""
+    x += rnd_direction.value[0]
+    y += rnd_direction.value[1]
+    return x, y
+
+def end_points(start_point:tuple)->list[tuple]:
+    """Creates a list of coords of legal end points"""
+    if start_point[0] == 0:
+        return [(Tile.WIDTH-1, y) for y in range(Tile.HEIGHT)]
+    elif start_point[1] == 0:
+        return [(x, Tile.HEIGHT-1) for x in range(Tile.WIDTH)]
+    elif start_point[0] == Tile.WIDTH-1:
+        return [(0, y) for y in range(Tile.HEIGHT)]
+    else:
+        return [(x, 0) for x in range(Tile.WIDTH)]
+
+def legal_move(x:int, y:int):
+    return 0<=x<Tile.WIDTH and 0<=y<Tile.HEIGHT
+
+def generate_level(lvl:int, difficulty:int):
```

Comment:
Jak se projevuje ten parametr **lvl** a **difficulty** pri generovani toho levelu? Ja to nikde nevidim.